edi_oca/models/edi_exchange_record.py

```diff
@@ -599,7 +626,37 @@ def check_access_rule(self, operation):
 
     def write(self, vals):
         self.check_access_rule("write")
-        return super().write(vals)
+        if "res_id" not in vals and "model" not in vals:
+            return super().write(vals)
+        # Updating main record information
+        res_id_updated = bool("res_id" in vals)
+        res_id = vals.get("res_id", False)
+        model_updated = bool("model" in vals)
+        model = vals.get("model", False)
+        related_records_command_list = [Command.clear()]
+        if (res_id_updated and not res_id) or (model_updated and not model):
+            # If res_id/model is emptied, we clear the related records
+            vals["related_record_ids"] = related_records_command_list
+            return super().write(vals)
+        if res_id_updated and res_id and model_updated and model:
+            # We can make bulk update
+            related_records_command_list.append(
+                Command.create({"res_id": res_id, "model": model})
+            )
+            vals["related_record_ids"] = related_records_command_list
+            return super().write(vals)
+        # Only updating res_id or model. We need to go one by one
```

Comment:
I'm not sure to understand why this is needed. Can you describe what is the expected behavior in general?
Eg: if you touch `res_id` or `model` you keep the old relation while replacing the main one?

IMO to keep things simple, you should always add a relation, either w/ an inverse on res_id or via `_set_related_records`. This way you don't need this complex machinery in write.

Then if you want to remove a record, it should be explicit, via a new method `_drop_related_records`.

WDYT?

Reply:
The goal is to make sure that custom logics that are currently working only with the main res_id and model keep working as expected. I was thinking on cases where any of these fields are changed, then the related records should also be updated to remove the old ones and add the new one. I agree that an inverse will be cleaner for this case.